File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        min_cost = []
        min_cost.append(cost[0])
        min_cost.append(cost[1])
        for i in range(2, len(cost)):
            min_cost.append(min(min_cost[i-1]+cost[i],min_cost[i-2]+cost[i]))
        n = len(cost)-1
        return min(min_cost[n],min_cost[n-1])
        

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> mini_cost;
        mini_cost.push_back(cost[0]);
        mini_cost.push_back(cost[1]);
        for(int i = 2; i < cost.size(); i++){
            int minx = min(mini_cost[i-1]+cost[i],mini_cost[i-2]+cost[i]);
            mini_cost.push_back(minx);
        }
        int n = cost.size()-1;
        return min(mini_cost[n],mini_cost[n-1]);
    }
};